frontend/ui/widget/inputdialog.lua

```diff
@@ -723,6 +723,76 @@ function InputDialog:_addScrollButtons(nav_bar)
                 end,
             })
         end
+        -- Add a button to go to the line by its number in the file
+        if self.fullscreen and self.view_pos_callback then
+            table.insert(row, {
+                text = "Go",
+                callback = function()
+                    -- calculate current line number
+                    local curr_line_num = 1
+                    for i = 1, #self._input_widget.charlist do
+                        if self._input_widget.charlist[i] == "\n" then
+                            if i >= self._input_widget.charpos then break end
+                            curr_line_num = curr_line_num + 1
+                        end
+                    end
+                    local last_line_num = select(2, string.gsub(self:getInputText(), "\n", "")) + 1
+                    input_dialog = InputDialog:new{
+                        title = _("Enter line number"),
+                        input_hint = T("%1 (1 - %2)", curr_line_num, last_line_num),
```

Comment:
The file is missing `local T = require("ffi/util").template`, so this currently crashes.